container/container_unix.go

```diff
@@ -283,6 +283,11 @@ func (container *Container) UpdateContainer(hostConfig *containertypes.HostConfi
 		cResources.CpusetMems = resources.CpusetMems
 	}
 	if resources.Memory != 0 {
+		// if memory limit smaller than already set memoryswap limit and doesn't
+		// update the memoryswap limit, then error out.
+		if resources.Memory > cResources.MemorySwap && resources.MemorySwap == 0 {
```

Comment:
Should this be 

```
if resources.Memory > cResources.MemorySwap &&  resources.Memory > resources.MemorySwap {
```

To check if a *new* MemorySwap is set, it's a valid new combination as well?